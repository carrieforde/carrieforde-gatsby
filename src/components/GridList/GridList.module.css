.gridList {
	display: grid;
	font-size: var(--font-size-2);
	grid-template-columns: repeat(auto-fit, minmax(175px, 1fr));
}

.gridListItem {
	margin-top: 0;
}

@media screen and (min-width: 600px) {
	.gridList {
		font-size: var(--font-size-3);
	}
}
