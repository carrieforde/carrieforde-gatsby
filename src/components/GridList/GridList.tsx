import List, { ListItem } from "components/List/List";
import React from "react";
import { gridList, gridListItem } from "./GridList.module.css";

const GridList: React.FC = ({ children }) => (
  <List type="BULLETED" className={gridList}>
    {children}
  </List>
);

const GridListItem: React.FC = ({ children }) => (
  <ListItem className={gridListItem}>{children}</ListItem>
);

export { GridList as default, GridListItem };
