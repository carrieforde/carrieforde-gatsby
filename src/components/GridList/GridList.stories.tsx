import { Meta, Story } from "@storybook/react";
import React from "react";
import GridList, { GridListItem } from "./GridList";

export default {
  title: "GridList",
  component: GridList,
} as Meta;

const Template: Story = (args) => <GridList {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: [
    <GridListItem key="Typescript">Typescript</GridListItem>,
    <GridListItem key="React">React</GridListItem>,
    <GridListItem key="HTML">HTML</GridListItem>,
    <GridListItem key="CSS (& Sass)">CSS (& Sass)</GridListItem>,
    <GridListItem key="Git">Git</GridListItem>,
    <GridListItem key="Firebase">Firebase</GridListItem>,
  ],
};
