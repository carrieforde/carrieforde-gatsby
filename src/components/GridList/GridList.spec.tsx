import { render } from "@testing-library/react";
import React from "react";
import { Default } from "./GridList.stories";

describe("GridList", () => {
  it("should render a bulleted GridList component", () => {
    const { container } = render(<Default {...Default.args} />);

    expect(container).toMatchSnapshot();
  });
});
