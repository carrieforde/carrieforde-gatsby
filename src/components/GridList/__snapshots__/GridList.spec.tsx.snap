// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridList should render a bulleted GridList component 1`] = `
<div>
  <ul
    class="gridList"
  >
    <li
      class="gridListItem"
    >
      Typescript
    </li>
    <li
      class="gridListItem"
    >
      React
    </li>
    <li
      class="gridListItem"
    >
      HTML
    </li>
    <li
      class="gridListItem"
    >
      CSS (& Sass)
    </li>
    <li
      class="gridListItem"
    >
      Git
    </li>
    <li
      class="gridListItem"
    >
      Firebase
    </li>
  </ul>
</div>
`;
